### PR TITLE
Allow usage of columns from multiple tables in the same order by expression

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

**Comment by matriv** [comment]
@seut Please review
The remainingOrderBy map is still on the RelationSplitter. If it's better to copy it and keep it in MultiSourceSelect please advise...

**Comment by matriv** [comment]
@seut Please review
The remainingOrderBy map is still on the RelationSplitter. Instead of copying to somewhere else like MultiSourceSelect I'm replacing its contents in the rewrite with map.clear(); map.putAll(newMap);
Please also advise for more comments in the code.

File: sql/src/main/java/io/crate/analyze/OrderBy.java [hunk 1]
```diff
@@ -160,4 +159,30 @@ public void replace(Function<? super Symbol, Symbol> replaceFunction) {
             listIt.set(replaceFunction.apply(listIt.next()));
         }
     }
+
+    @Nullable
+    public static OrderBy mergeOrderBy(OrderBy orderBy1, OrderBy orderBy2) {
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/OrderBy.java]: maybe change to non static one and rename to just `merge`

Comment (seut) [inline sql/src/main/java/io/crate/analyze/OrderBy.java]: also maybe allow orderby to merge to be nullable and return this if its nullable

File: sql/src/main/java/io/crate/analyze/relations/RelationSplitter.java [hunk 2]
```diff
@@ -161,15 +160,23 @@ private void processOrderBy() {
...
                 // not pushed down
                 splits.put(null, idx);
+                // Create remainingOrderBy for the expressions that are not pushed down
+                OrderBy newOrderBy = orderBy.subset(Arrays.asList(idx));
+                Set<QualifiedName> qualifiedNames = new HashSet<>(relations.size());
+                for (AnalyzedRelation rel : relations) {
+                    qualifiedNames.add(rel.getQualifiedName());
+                }
+                OrderBy existingOrderBy = remainingOrderBy.get(qualifiedNames);
+                remainingOrderBy.put(qualifiedNames, existingOrderBy == null ? newOrderBy :
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/relations/RelationSplitter.java]: change this after changing orderBy.merge() according to my related comments

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 3]
```diff
@@ -249,6 +248,27 @@ public Symbol apply(@Nullable Symbol input) {
         return newMap;
     }
 
+    private static void rewriteOrderByNames(Map<Set<QualifiedName>, OrderBy> remainingOrderBy,
```
Comment (seut) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: can we maybe share the same code for query and order by by using generics?

File: sql/src/test/java/io/crate/planner/fetch/MultiSourceFetchPushDownTest.java [hunk 4]
```diff
@@ -90,11 +90,17 @@ public void testPushDownWithOrder() throws Exception {
 
     @Test
     public void testPushDownWithMultiRelationOrder() throws Exception {
-        pushDown("select a, b from t1, t2 order by x - y");
+        pushDown("select a, b from t1, t2, t3 order by x - y + z");
```
Comment (seut) [inline sql/src/test/java/io/crate/planner/fetch/MultiSourceFetchPushDownTest.java]: is this test really testing the MultiSourceFetchPushDown? isn't most logic done at the RelationSplitter now?
if so, please move the test and if some fetch push down related is left over here, modify and rename the test here

File: CHANGES.txt [hunk 5]
```diff
@@ -5,6 +5,10 @@ Changes for Crate
 Unreleased
 ==========
 
+ - Remove limitation that prevented usage of columns from multiple tables
+   in the same order by expression.
+   eg: select t1.x, t2.y, t3.y from t1, t2, t3 order by t1.x - t2.y + t3.z
```
Comment (seut) [inline CHANGES.txt]: afaik an order by example without select and from is enough, also maybe use capital letters

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 6]
```diff
@@ -248,25 +250,26 @@ public Symbol apply(@Nullable Symbol input) {
         return newMap;
     }
 
-    private static void rewriteOrderByNames(Map<Set<QualifiedName>, OrderBy> remainingOrderBy,
-                                            QualifiedName leftName,
-                                            QualifiedName rightName,
-                                            QualifiedName newName,
-                                            Function<? super Symbol, Symbol> replaceFunction) {
+    private static Map<Set<QualifiedName>, OrderBy> rewriteOrderByNames(Map<Set<QualifiedName>, OrderBy> remainingOrderBy,
+                                                                    QualifiedName leftName,
```
Comment (seut) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: check idention

File: sql/src/test/java/io/crate/analyze/relations/RelationSplitterTest.java [hunk 7]
```diff
@@ -187,6 +188,27 @@ public void testSplitOrderByWith3RelationsButOutputsOnly2Relations() throws Exce
...
+    public void testSplitOrderByCombiningColumnsFrom3Relations() throws Exception {
+        // select a, b from t1, t2, t3 order by x - y + z
+        QuerySpec querySpec = new QuerySpec().outputs(Arrays.asList(asSymbol("a"), asSymbol("b")));
+        List<Symbol> orderBySymbols = Arrays.asList(asSymbol("x - y + x"));
+        OrderBy orderBy = new OrderBy(orderBySymbols, new boolean[]{false}, new Boolean[]{null});
+        querySpec.orderBy(orderBy);
+
+        RelationSplitter splitter = split(querySpec);
+
+        assertThat(querySpec, isSQL("SELECT doc.t1.a, doc.t2.b ORDER BY add(subtract(doc.t1.x, doc.t2.y), doc.t1.x)"));
```
Comment (seut) [inline sql/src/test/java/io/crate/analyze/relations/RelationSplitterTest.java]: pls check, T3 is not used here

**Comment by seut** [comment]
lgtm

**Pull request merged**